e2e: serial: create workload with TAS followed by default scheduler

swatisehgal opened the pull request:
Create a workload using TAS scheduler that is scheduled followed by
another similar workload using the default scheduler that stays pending.

Signed-off-by: Swati Sehgal <swsehgal@redhat.com>

fromanirh commented:
CI failures look legit
```
# github.com/openshift-kni/numaresources-operator/test/e2e/serial/tests
Error: test/e2e/serial/tests/workload_placement.go:845:28: baseload.CPU undefined (type baseload.Load has no field or method CPU)
Error: test/e2e/serial/tests/workload_placement.go:848:25: baseload.CPU undefined (type baseload.Load has no field or method CPU)
Error: test/e2e/serial/tests/workload_placement.go:851:28: baseload.Memory undefined (type baseload.Load has no field or method Memory)
Error: test/e2e/serial/tests/workload_placement.go:854:25: baseload.Memory undefined (type baseload.Load has no field or method Memory)
make: *** [Makefile:114: vet] Error 2
Error: Process completed with exit code 2.
```

fromanirh commented:
I'd be in favour of adding `CPU()` and  `Memory` helpers to `baseload.Load` if this helps

fromanirh commented on internal/baseload/baseload.go:
```diff
@@ -103,3 +103,11 @@ func (nl Load) ToResourceList() corev1.ResourceList {
 	return res
 }
+
+func (nl Load) CPULoad() resource.Quantity {
```
I'm still very much in favor of adding these, but I'm surprised github is not complaining - we merged https://github.com/openshift-kni/numaresources-operator/pull/361 which changed the internal representation, so this PR actually needs a rebase

swatisehgal replied:
Yep, on it.

swatisehgal commented:
Updates:
1. Implemented this test case with a pod instead of a deployment.
2. Determined resources available per NUMA node and used them as a request for the two containers instead of the implementation eliminating the need for `leastAvailableResourceQtyInAllZone` and `FindResourceAllocatableByname`. That inturn meant that there was no need to expose baseload CPU and memory. These changes would be needed and implemented in https://github.com/openshift-kni/numaresources-operator/pull/327.
3. This PR was tested locally and is passing.

fromanirh reviewed (commented):
/lgtm
/approve

fromanirh commented:
/retest
